sdcm/ndbench_thread.py

```diff
@@ -0,0 +1,234 @@
+
+    def __init__(self, loader_node, loader_idx, ndbench_log_filename):
+        super(NdBenchStatsPublisher, self).__init__()
+        self.loader_node = loader_node
+        self.loader_idx = loader_idx
+        self.ndbench_log_filename = ndbench_log_filename
+
+        for operation in self.collectible_ops:
+            gauge_name = self.gauge_name(operation)
+            if gauge_name not in self.METRICS:
+                metrics = nemesis_metrics_obj()
+                self.METRICS[gauge_name] = metrics.create_gauge(gauge_name,
+                                                                'Gauge for ndbench metrics',
+                                                                ['instance', 'loader_idx', 'type'])
+
+    @staticmethod
+    def gauge_name(operation):
+        return 'collectd_ndbench_%s_gauge' % operation.replace('-', '_')
+
+    def set_metric(self, operation, name, value):
+        metric = self.METRICS[self.gauge_name(operation)]
+        metric.labels(self.loader_node.ip_address, self.loader_idx, name).set(value)
+
+    def run(self):
+        # INFO RPSCount:78 - Read avg: 0.314ms, Read RPS: 7246, Write avg: 0.39ms, Write RPS: 1802, total RPS: 9048, Success Ratio: 100%
+        stat_regex = re.compile(
+            r'Read avg: (?P<read_lat_avg>.*?)ms.*?Read RPS: (?P<read_ops>.*?),.*?Write avg: (?P<write_lat_avg>.*?)ms.*?Write RPS: (?P<write_ops>.*?),', re.IGNORECASE)
+
+        while not self.stopped():
+            exists = os.path.isfile(self.ndbench_log_filename)
+            if not exists:
+                time.sleep(0.5)
+                continue
+
+            for _, line in enumerate(self.follow_file(self.ndbench_log_filename)):
+                if self.stopped():
+                    break
+                try:
+                    match = stat_regex.search(line)
+                    if match:
+                        for key, value in match.groupdict().items():
+                            operation, name = key.split('_', 1)
+                            self.set_metric(operation, name, float(value))
+
+                except Exception:  # pylint: disable=broad-except
+                    LOGGER.exception("fail to send metric")
+
+
+def convert_bool_or_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        pass
+
+    try:
+        return strtobool(value)
+    except ValueError:
+        pass
+
+    return value
+
+
+class RemoteDocker:
+    def __init__(self, node, image_name, ports=None, command_line="tail -f /dev/null", extra_docker_opts=""):  # pylint: disable=too-many-arguments
+        self.node = node
+        self._internal_ip_address = None
+
+        ports = " ".join([f'-p {port}:{port}' for port in ports]) if ports else ""
+        res = self.node.remoter.run(
+            f'docker run {extra_docker_opts} -d {ports} {image_name} {command_line}', verbose=True)
+        self.docker_id = res.stdout.strip()
+        atexit.register(self.atexit)
```

Comment:
remove and add a new method that stops the docker container, call this method as soon as the container not needed

Reply:
removed